### PR TITLE
No ticket/add push notifications

### PR DESCRIPTION
**Pull request by cmjarreau**
starter push notifications example

https://user-images.githubusercontent.com/3827284/232093454-50590c1f-9896-4e1a-91e5-6598dcea51db.mov

### CODE REVIEW

File: src/Screens/PushNotificationsScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      //@ts-ignore
+      silent: false,
+      category: 'SOME_CATEGORY',
+      userInfo: {},
+    });
+  };
+
+  useEffect(() => {
+    requestNotificationsPermissions(({ deviceToken, denied, error }) => {
+      return { deviceToken, denied, error };
```
Comment (cmjarreau) [inline src/Screens/PushNotificationsScreen.tsx:45]: for a lot of these return statements, im not exactly sure what we want to do with these values. or if we even want to return them at all and instead log them.

File: src/App.tsx [hunk 2]
```diff
@@ -1,7 +1,16 @@
...
 import { RootProviders, RootStack, init } from '@lifeomic/react-native-sdk';
+import { PushNotificationsScreen } from './Screens/PushNotificationsScreen';
+
+const PushNotificationScreenProvider = ({
+  children,
+}: {
+  children?: React.ReactNode;
+}) => {
+  return <PushNotificationsScreen />;
+};
```
Comment (evanheisler) [inline src/App.tsx:13]: I'm a bit confused about this. Maybe you could clarify:

* This isn't really a provider. There isn't any shared context, state, or functionality being "provided" to its children
* This takes props, but doesn't do anything with them. It is just rendering a component and preventing the `RootStack` from rendering which seems like a problem.

File: src/App.tsx [hunk 3]
```diff
@@ -21,7 +30,9 @@ const authConfig: AuthConfiguration = {
 function App() {
   return (
     <RootProviders authConfig={authConfig}>
-      <RootStack />
+      <PushNotificationScreenProvider>
+        <RootStack />
+      </PushNotificationScreenProvider>
```
Comment (evanheisler) [inline src/App.tsx:35]: As mentioned above, I don't think this is the correct implementation.

If the end goal here is to have someone copy this repo and have a notifications "playground", I think you have to create a new navigation stack to render your `PushNotificationsScreen` alongside the RootStack. But from a consumer standpoint, this doesn't really make sense. You would never view notifications without being signed in.

File: src/Screens/PushNotificationsScreen.tsx (hunk 1 above)
Comment (evanheisler) [reply src/Screens/PushNotificationsScreen.tsx:45]: I imagine we want to provide an example implementation. Such as (pseudo code)
```typescript
if (error || denied) {
  setError(error) // See the SDK for showing a <Toast /> message
}

// Register device with application 
api.post('/v1/device-endpoints'. { deviceToken, provider: 'device platform, e.g. apns or gcm', application: 'my-app' }
```

**Review by evanheisler — CHANGES_REQUESTED** [review]

**Comment by evanheisler** [comment]
It seems to me that these changes are more akin to a storybook story that belong in the SDK, demonstrating how one would build out their apps' notifications. The role of the starter would be pre-configuration to leverage some built-in functionality.

**Pull request closed without merging**